model/workload/workload.go

```diff
@@ -4,38 +4,69 @@ package workload
 
 import (
 	"fmt"
-	log "github.com/sirupsen/logrus"
 	"net/http"
 	"reflect"
 	"strconv"
 	"strings"
 	"sync"
 
+	rmderror "github.com/intel/rmd/api/error"
+	"github.com/intel/rmd/db"
 	syscache "github.com/intel/rmd/lib/cache"
 	"github.com/intel/rmd/lib/cpu"
+	l_mba "github.com/intel/rmd/lib/mba"
 	"github.com/intel/rmd/lib/proc"
 	"github.com/intel/rmd/lib/proxyclient"
 	"github.com/intel/rmd/lib/resctrl"
 	libutil "github.com/intel/rmd/lib/util"
-
-	rmderror "github.com/intel/rmd/api/error"
-	"github.com/intel/rmd/db"
 	"github.com/intel/rmd/model/cache"
+	m_mba "github.com/intel/rmd/model/mba"
 	"github.com/intel/rmd/model/policy"
 	tw "github.com/intel/rmd/model/types/workload"
 	"github.com/intel/rmd/util"
 	"github.com/intel/rmd/util/rdtpool"
 	rmdbase "github.com/intel/rmd/util/rdtpool/base"
+	log "github.com/sirupsen/logrus"
 )
 
 var l sync.Mutex
 
+type mixedCandidate struct {
+	bitmap *libutil.Bitmap
+	mba    *string
+}
+
 // Validate the request workload object is validated.
-func Validate(w *tw.RDTWorkLoad) error {
+func Validate(w *tw.RDTWorkLoad, mbaInfo *m_mba.Info) error {
 	if len(w.TaskIDs) <= 0 && len(w.CoreIDs) <= 0 {
 		return fmt.Errorf("No task or core id specified")
 	}
 
+	// Check if user set MbaMbps
+	if w.MbaMbps != nil {
+		return fmt.Errorf("MBA Mbps didn't implemented yet")
+	}
+
+	// workloads API only accepts mba configuration for guaranteed pool
+	if w.MbaPercentage != nil || w.MbaMbps != nil {
+		if mbaInfo.MbaOn == false {
+			return fmt.Errorf("This platform does not support MBA")
+		}
+		if w.MaxCache != nil && w.MinCache != nil {
+			if (*w.MaxCache == 0 && *w.MinCache == 0) ||
+				(*w.MaxCache > 0 && *w.MinCache > 0 && *w.MaxCache > *w.MinCache) {
+				return fmt.Errorf("Setting MBA on shared pool is forbidden")
+			}
+		}
+	}
+
+	if w.MbaPercentage != nil {
+		if *w.MbaPercentage > 100 || (int)(*w.MbaPercentage) < mbaInfo.MbaMin ||
+			(int)(*w.MbaPercentage)%mbaInfo.MbaStep != 0 {
```

Comment:
I do not think Mba step check is needed. It is an internal control mechanism.

root@NC-SKL:/sys/fs/resctrl/test# echo "MB:0=20;1=100" > schemata 
root@NC-SKL:/sys/fs/resctrl/test# cat schemata 
    L3:0=7ff;1=7ff
    MB:0= 20;1=100
root@NC-SKL:/sys/fs/resctrl/test# echo "MB:0=21;1=100" > schemata 
root@NC-SKL:/sys/fs/resctrl/test# cat schemata 
    L3:0=7ff;1=7ff
    MB:0= 30;1=100

As you can see, the system automatically rounds it off for the user and this will not have much impact.